### PR TITLE
[stdlib] DoubleWidth refinements

### PR DESCRIPTION
**Pull request by natecook1000**
This adds tests for masking bit shifts to `DoubleWidth` and addresses the feedback from @dabrahams here: https://github.com/apple/swift/pull/10131

### CODE REVIEW

**Comment by natecook1000** [comment]
@swift-ci Please test

**Comment by natecook1000** [comment]
@swift-ci Please test OS X platform

**Comment by natecook1000** [comment]
@swift-ci Please test OS X platform

**Comment by moiseev** [comment]
Can we please wait till https://github.com/apple/swift/pull/11203 lands? Please?

File: stdlib/public/core/DoubleWidth.swift.gyb [hunk 1]
```diff
@@ -481,32 +483,30 @@ public struct DoubleWidth<Base : FixedWidthInteger> :
...
     lhs._storage.high <<= High(rhs._storage.low)
-    if Base.bitWidth > rhs._storage.low {
-      lhs._storage.high |= High(extendingOrTruncating: lhs._storage.low >>
-        (numericCast(Base.bitWidth) - rhs._storage.low))
-    } else {
-      lhs._storage.high |= High(extendingOrTruncating: lhs._storage.low <<
-        (rhs._storage.low - numericCast(Base.bitWidth)))
-    }
+
+    let t = Base.bitWidth > rhs._storage.low
```
Comment (moiseev) [inline stdlib/public/core/DoubleWidth.swift.gyb]: `t` for temporary?

**Review by moiseev — CHANGES_REQUESTED** [review]
LGTM, apart from the `t` variable name. Maybe `newLow` and `newHigh` would read better?

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

**Review by moiseev — APPROVED** [review]
:shipit:

**Pull request merged**